src/app/modules/authController.js

```diff
@@ -14,16 +30,42 @@ class UserController {
       password: hashPassword,
     };
     const result = await UserServices.create(authData);
-    const { userId } = result.dataValues;
-    const token = jwt.createToken(userId);
 
+    const uID = req.body.username;
+    const getId = await UserServices.findUserid(uID);
+
+    const { id } = getId;
+    const token = jwt.createToken(id);
     if (!result) {
```

Comment:
In your code you are just passing an object containing the password. You should consider storing all the information into the database not only the password. So after creating a user, the result is stored into the the variable **result**. Then you can use that object to generate the token because it should contain all those information. Then you can destructure those information to get the **id, username & email**. Clean your code...

Reply:
everything is being stored in the database , maybe just that the token should be generated based on whole result